python/plugins/processing/algs/qgis/PointsToPaths.py

```diff
@@ -73,8 +76,8 @@ def initAlgorithm(self, config=None):
                                                               self.tr('Input point layer'), [QgsProcessing.TypeVectorPoint]))
         self.addParameter(QgsProcessingParameterBoolean(self.CLOSE_PATH,
                                                         self.tr('Close path'), defaultValue=False))
-        self.addParameter(QgsProcessingParameterField(self.ORDER_FIELD,
-                                                      self.tr('Order field'), parentLayerParameterName=self.INPUT))
+        self.addParameter(QgsProcessingParameterExpression(self.ORDER_FIELD,
```

Comment:
Unfortunately this could potentially break models/scripts -- e.g. if a script is using a hardcoded field name with a space in it, then it won't automatically become a valid QGIS expression. I think you'll need to make a new parameter here and hide the old one.